scripts/prepare_slimpajama.py

```diff
@@ -0,0 +1,62 @@
+import json
+import os
+import sys
+import time
+from pathlib import Path
+
+import zstandard as zstd
+from lightning.data.streaming import DataProcessor, DataChunkRecipe
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt import Tokenizer
+
+
+class SlimPajamaDataRecipe(DataChunkRecipe):
+    def __init__(self, tokenizer: Tokenizer, chunk_size: int):
+        super().__init__(chunk_size)
+        self.tokenizer = tokenizer
+
+    def prepare_structure(self, input_dir):
+        files = Path(input_dir).rglob("*.zst")
+        return [str(file) for file in files]
+
+    def prepare_item(self, filepath):
+        with zstd.open(open(filepath, "rb"), "rt", encoding="utf-8") as f:
```

Comment:
Isn't it strange to open an opened file? Shouldn't we pass the filepath directly to zstd? https://python-zstandard.readthedocs.io/en/latest/misc_apis.html#open

Reply:
the files are compressed, so they need to be read in binary. This binary handle is then passed to zstandard. Even though the zstandard.open() argument is named filename, it can actually take the binary file io.